build.sh

```diff
@@ -1,37 +1,77 @@
-set -e 
+#!/bin/sh
+# Build script of python package
+
+set -e
 name="oval-graph"
 module="oval_graph"
 
+unit_test() {
+    python3 -m pytest
+}
+
+commit_version() {
+    version=$1
+    module=$2
+    git add ${module}/__init__.py
+    git add oval-graph.spec
+    git commit -m "${version}"
+    git tag "${version}"
+    git push --follow-tags
+    git push upstream master
+}
+
+install_build_requirements() {
+    python3 -m pip install --user --upgrade setuptools wheel twine
+}
+
+build_backage() {
+    rm -rf dist
+    python3 setup.py sdist bdist_wheel
+    twine upload dist/*
+}
+
+update_version_package() {
+    module=$1
+    sed -i "s/$old_version/$new_version/g" ${module}/__init__.py
+}
+
+update_version_rpm() {
+    version=$1
+    name=$2
+    rpmdev-bumpspec "${name}".spec --comment="release ${version}" -n "${version}" --userstring="Jan Rodak <jrodak@redhat.com>"
+}
+
+load_upstream() {
+    git checkout master
+    git pull upstream master
+}
+
 if [ "$1" != "" ]; then
     new_version=$1
-    if [[ $new_version =~ ^[0-9]{1,}.[0-9]{1,}.[0-9]{1,}$ ]]; then
+    if echo "$new_version" | grep -q -E '^[0-9]{1,}.[0-9]{1,}.[0-9]{1,}$'; then
         old_version=$(python3 setup.py --version)
-        git checkout master
-        git pull upstream master
+        load_upstream
         if git status --porcelain=v1 | grep -q '^\(.M\|M.\)'; then
             echo "Not commited changes!"
         else
             # update version in file
-            sed -i "s/$old_version/$new_version/g" ${module}/__init__.py
+            update_version_package "$module"
             version=$(python3 setup.py --version)
-            rpmdev-bumpspec oval-graph.spec --comment="release ${version}" -n "${version}" --userstring="Jan Rodak <jrodak@redhat.com>"
-            if [ "$version" == "$new_version" ]; then
+            update_version_rpm "$version" "$name"
+            if [ "$version" = "$new_version" ]; then
                 # Commit version
-                git add ${module}/__init__.py
-                git add oval-graph.spec   
-                git commit -m "${version}"
-                git tag "${version}"
-                git push --follow-tags
-                git push upstream master
-                # Build  
-                rm -rf dist
-                python3 setup.py pytest sdist bdist_wheel
-                twine upload dist/*
+                commit_version "$version" "$module"
```

Comment:
That would be the last thing to do - first make the change to the version, then go all the way to tests, then commit, and only when you feel confident push the tag. You can undo pushing a tag, but that should be an absolute exception.